### PR TITLE
fix(core): Load WEBHOOK_URL via config to support setting the value via a file

### PR DESCRIPTION
**Pull request by moredhel**
Fix mismatch between configuration between environment variables and json-file configuration. Updates references to WEBHOOK_URL to refer to `config.get('endpoints.webhookUrl')`. Updates the schema with endpoints.webhookUrl

Github issue / Community forum post (link here to close automatically):

### CODE REVIEW

File: packages/cli/src/commands/start.ts [hunk 1]
```diff
@@ -347,8 +348,11 @@ export class Start extends BaseCommand {
 			// @ts-ignore
 			const webhookTunnel = await localtunnel(port, tunnelSettings);
 
-			process.env.WEBHOOK_URL = `${webhookTunnel.url}/`;
-			this.log(`Tunnel URL: ${process.env.WEBHOOK_URL}\n`);
+			// Change required here
+			config.set('endpoints.webhookUrl', webhookTunnel.url);
```
Comment (netroy) [inline packages/cli/src/commands/start.ts:352]: shouldn't this be 
```suggestion
			config.set('endpoints.webhookUrl', `${webhookTunnel.url}/`);
```
to be backward compatible?

Comment (moredhel) [reply packages/cli/src/commands/start.ts]: You are correct, I've accepted your change.

Thanks for having a look :)

**Review by netroy — COMMENTED** [review]
we have quite a few places in code that fetch the webhook base url, and then cache it. which means that changing the value in value in `env` or in `config` doesn't update all these places.
To fix this we need to either update the code to setup the tunnel before any code requests  `getWebhookBaseUrl`, or switch to a reactive pattern to ensure that all references to this url are updated across the code.

**Comment by netroy** [comment]
marking this as draft, as we need to fundamentally change how this url is passed around in the code.

**Comment by netroy** [comment]
@moredhel thanks for the PR, but this turned out to be a more involved change than I had anticipated. this issue will be addressed in https://github.com/n8n-io/n8n/pull/6713 instead.

**Pull request closed without merging**